cmd/sniproxy/snipproxy.go

```diff
@@ -23,6 +24,7 @@ import (
 )
 
 var ports = flag.String("ports", "443", "comma-separated list of ports to proxy")
+var promoteHTTPS = flag.Bool("promote-https", false, "promote HTTP to HTTPS")
```

Comment:
I'm not even sure it needs a flag, seems like always the right behavior. At least I'd default to true.